### PR TITLE
Fix HyperUniquesAggregatorFactory.estimateCardinality null handling to respect output type

### PR DESCRIPTION
**Pull request by maytasm**
Make return type consistent for HyperUniquesAggregator and HyperUniquesVectorAggregator when no value found

### Description

This bug was discovered by turning on vectorization. In HyperUniquesAggregator, when a column is empty, `collector` will be null and the get() method will returns null. In HyperUniquesVectorAggregator, when a column is empty, get() method will still returns a HyperLogLogCollector (from the call to `HyperUniquesBufferAggregator.doGet(buf, position);`). In HyperUniquesAggregatorFactory#estimateCardinality (called by HyperUniquesAggregatorFactory#finalizeComputation) we return integer 0 if input is null, while we return long/double value. otherwise. Hence, for the above mentioned case, the non-vectorize returns integer 0 while the vectorize returns long/double 0. The test failed since we are expecting integer 0. 
This PR, make sure that for the case when the input to HyperUniquesAggregatorFactory#estimateCardinality is null, we still return 0 with the same value type as when it is not null.  

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/QueryContexts.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
   public static final boolean DEFAULT_USE_CACHE = true;
   public static final boolean DEFAULT_POPULATE_RESULTLEVEL_CACHE = true;
   public static final boolean DEFAULT_USE_RESULTLEVEL_CACHE = true;
-  public static final Vectorize DEFAULT_VECTORIZE = Vectorize.FALSE;
+  public static final Vectorize DEFAULT_VECTORIZE = Vectorize.TRUE;
```
Comment (maytasm) [inline processing/src/main/java/org/apache/druid/query/QueryContexts.java:62]: Just to see if anything else fail. Will make this change in a separate PR with relevant doc change

**Comment by clintropolis** [comment]
>FWIW I like that the result type will be consistent after this change, and I can't think of any computations going awry if the returned value is 0d instead of 0L

I consider the existing behavior a bug, it should have always been respecting the output type. Using one of the tests that had to change, it didn't really make sense that null handling mode should change the output type.
